tools/perf_util/src/main/java/com/msft/ort_perf_tool/App.java

```diff
@@ -0,0 +1,179 @@
+import java.sql.Types;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+public class App {
+
+	static String exec_command(Path source_dir, String... commands) throws Exception {
+		ProcessBuilder sb = new ProcessBuilder(commands).directory(source_dir.toFile()).redirectErrorStream(true);
+		Process p = sb.start();
+		if (p.waitFor() != 0)
+			throw new RuntimeException("execute " + String.join(" ", commands) + " failed");
+		try (BufferedReader r = new BufferedReader(new InputStreamReader(p.getInputStream()))) {
+			return r.readLine();
+		}
+	}
+
+	public static void main(String[] args) throws Exception {
+
+		final Path source_dir = Paths.get(args[0]);
+		final List<Path> perf_metrics = new ArrayList<Path>();
+		Files.walkFileTree(source_dir, new SimpleFileVisitor<Path>() {
+
+			@Override
+			public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) throws IOException {
+				String dirname = dir.getFileName().toString();
+				if (dirname != "." && dirname.startsWith("."))
+					return FileVisitResult.SKIP_SUBTREE;
+				return FileVisitResult.CONTINUE;
+			}
+
+			@Override
+			public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+				String filename = file.getFileName().toString();
+
+				if (!filename.startsWith(".") && filename.endsWith(".json")) {
+					perf_metrics.add(file);
+				}
+				return FileVisitResult.CONTINUE;
+			}
+
+		});
+		System.out.println(perf_metrics);
+
+		final Path cwd_dir = Paths.get(System.getProperty("user.dir"));
+		// git rev-parse HEAD
+		String commit_id = exec_command(cwd_dir, "git", "rev-parse", "HEAD");
+		String date = exec_command(cwd_dir, "git", "show", "-s", "--format=%ci", commit_id);
+		final SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss Z");
+		java.util.Date commitDate = sdf.parse(date);
+		final SimpleDateFormat simple_date_format = new SimpleDateFormat("yyyy-MM-dd");
+		String batch_id = simple_date_format.format(commitDate);
+		System.out.println(String.format("Commit change date: %s", batch_id));
+
+		// collect all json files list
+		processPerfMetrics(source_dir, perf_metrics, commit_id, batch_id);
+
+		// TODO - add e2e tests later, run it w/ process command
+	}
+
+	private static void processPerfMetrics(Path source_dir, final List<Path> perf_metrics, String commit_id,
+										   String batch_id) throws Exception {
+		try {
+			Connection conn = JdbcUtil.GetConn();
+			// go thru each json file
+			JSONParser jsonParser = new JSONParser();
+			for (Path metrics_json : perf_metrics) {
+				try (FileReader reader = new FileReader(metrics_json.toAbsolutePath().toString())) {
+					// Read JSON file
+					Object obj = jsonParser.parse(reader);
+					loadMetricsIntoMySQL(conn, commit_id, batch_id, (JSONObject) obj);
+				}
+			}
+		} finally {
+			deletePath(source_dir);
+		}
+	}
+
+	private static void deletePath(Path temp_path) throws IOException {
```

Comment:
why do we want to delete the files? the CI build can clean up the whole agent directory at the end.